test/partiql-tests-runner/src/test/kotlin/org/partiql/runner/executor/EvalExecutor.kt

```diff
@@ -54,11 +58,34 @@ class EvalExecutor(
 
     override fun compare(actual: PartiQLResult, expect: PartiQLResult): Boolean {
         if (actual is PartiQLResult.Value && expect is PartiQLResult.Value) {
-            return actual.value == expect.value
+            return valueComparison(actual.value, expect.value)
         }
         error("Cannot compare different types of PartiQLResult")
     }
 
+    // Value comparison of PartiQL Value that utilized Ion Hashcode.
+    // in here, null.bool is considered equivalent to null
+    // missing is considered different from null
+    // annotation::1 is considered different from 1
+    // 1 of type INT is considered the same as 1 of type INT32
+    // we should probably consider adding our own hashcode implementation
+    private fun valueComparison(v1: PartiQLValue, v2: PartiQLValue): Boolean {
+        if (v1.isNull && v2.isNull) {
+            return true
+        }
```

Comment:
`someAnnotation::null.bool` == `someAnnotation::null.int`

`someAnnotation::null.bool` != `otherAnnotation::null.int`

Reply:
I've already discussed with @yliuuuu, but for others who may be reading this comment, the KDoc says `annotation::1 is considered different from 1` and `null.bool is considered equivalent to null`. However, this implementation doesn't handle annotations on nulls. It should be a quick fix.

Reply:
Nice call out : )